src/hotspot/share/opto/doCall.cpp

```diff
@@ -239,7 +241,8 @@ CallGenerator* Compile::call_generator(ciMethod* callee, int vtable_index, bool
           // this invoke because it may lead to bimorphic inlining which
           // a speculative type should help us avoid.
           receiver_method = callee->resolve_invoke(jvms->method()->holder(),
-                                                   speculative_receiver_type);
+                                                   speculative_receiver_type,
+                                                   check_access);
```

Comment:
Can you explain why only here you pass `check_access` and expect it is `true` in all other places?

Reply:
> Can you explain why only here you pass check_access and expect it is true in all other places?

@vnkozlov That's the only case which was overlooked in JDK-8062280. All other cases aren't exercised for MH intrinsic methods and the asserts are there to verify that. If they start to fail, it'll signal that there may be a missing optimization opportunity.

> should we add an assert for check_access before the resolve_invoke in Compile::optimize_inlining?

@liach good question, it makes sense to separately take a closer look at this particular case. My first impression is `check_access` should be passed into ` resolve_invoke` rather than asserting `check_access == true` before `resolve_invoke`.